src/environmentbase/environmentbase.py

```diff
@@ -593,8 +607,12 @@ def initialize_template(self):
         self.template.add_utility_bucket(name=bucket_name)
         self.template.add_output(Output('utilityBucket',Value=bucket_name))
 
-        ami_cache = res.load_yaml_file(self.config.get('template').get('ami_map_file'))
-        self.template.add_ami_mapping(ami_cache)    
+        ami_filename = self.config['template']['ami_map_file']
+        ami_cache = res.load_yaml_file(ami_filename)
+
+        region_name = self.config['boto']['region_name']
+        restricted_ami_cache = {region_name: ami_cache[region_name]}
```

Comment:
This is not good if we are trying to generate templates that can be deployed in multiple regions right?